src/mlpack/core/data/random_regression_generator.hpp

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file random_regression_generator.hpp
+ * @author Ali Hossam
+ *
+ * Implementation of a regression data generator with random features and error 
+ * distribution.
+ *
```

Comment:
I am not sure, this is not a method, it is more of a simulation, so I have no idea if we need to add this to mlpack or not, it is nice to have, but I do not know where we can add this, @rcurtin would know better than me to judge this class in itself and whether if it is worth it or not